pkg/storage/azurecdn/storage.go

```diff
@@ -0,0 +1,87 @@
+package azurecdn
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/Azure/azure-storage-blob-go/2017-07-29/azblob"
+)
+
+// Storage implements (github.com/gomods/athens/pkg/storage).Saver and
+// also provides a function to fetch the location of a module
+type Storage struct {
+	accountURL *url.URL
+	cred       azblob.Credential
+}
+
+// New creates a new azure CDN saver
+func New(accountName, accountKey string) (*Storage, error) {
+	u, err := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net", accountName))
+	if err != nil {
+		return nil, err
+	}
+	cred := azblob.NewSharedKeyCredential(accountName, accountKey)
+	return &Storage{accountURL: u, cred: cred}, nil
+}
+
+// BaseURL returns the base URL that stores all modules. It can be used
+// in the "meta" tag redirect response to vgo.
+//
+// For example:
+//
+//	<meta name="go-import" content="gomods.com/athens mod BaseURL()">
+func (s Storage) BaseURL() *url.URL {
+	return s.accountURL
+}
+
+// Save implements the (github.com/gomods/athens/pkg/storage).Saver interface.
+func (s *Storage) Save(module, version string, mod, zip, info []byte) error {
+	ctx := context.Background()
+
+	pipe := azblob.NewPipeline(s.cred, azblob.PipelineOptions{})
+	serviceURL := azblob.NewServiceURL(*s.accountURL, pipe)
+	// according to the first example in:
+	// https://godoc.org/github.com/Azure/azure-storage-blob-go/2017-07-29/azblob
+	// ... container names need to be lower case
+	containerName := strings.ToLower(fmt.Sprintf("%s/@v", module))
```

Comment:
There are few more rules about container names, you may want to take a look here:
https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names

Reply:
oh man, that kinda sucks. I'll use a static container name and make the blob URLs fancy